lib/utils/appdb_reader.rb

```diff
@@ -17,36 +17,44 @@ def appdb_raw_request
 
       def all(options = {})
         response = options[:appdb_raw] || appdb_raw_request
-        if response.success?
-          response = response.parsed_response
-          response['appdb']['site'].collect do |site|
-            [site['service']].flatten.collect do |service|
-              {
-                id: "#{site['id']}:#{service['id']}",
-                name: site['name'],
-                country: site['country']['isocode'],
-                endpoint: service['occi_endpoint_url'],
-                site_id: site['id'],
-                service_id: service['id'],
-                flavours: filter_flavour([service['template']].flatten),
-                appliance: get_appliances(service['image'])
-              } if service['type'] == 'occi'
-            end
-          end.flatten.compact
-        end
+        response.success? ? parse_appdb_response(response.parsed_response) : nil
```

Comment:
Not relevant, nil means there was an error. Empty response can be generated as from parse_appdb_response function